### PR TITLE
[WIP] TF tasks wrapped into job 

### PR DESCRIPTION
**Pull request by snripa**

### CODE REVIEW

File: libs/infrastructure.py [hunk 1]
```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
             tfvars.write('{} = "{}"\n'.format("kube_nodes_amount", self.kube_nodes_amount))
             tfvars.write('{} = "{}"\n'.format("kube_nodes_instance_type", self.kube_nodes_instance_type))
 
-    def set_aws_cli_config(self):
+    def __set_aws_cli_config(self):
+        result = 0
         process = Popen(["aws", "configure", "set", "aws_access_key_id",
```
Comment (ronenhamias) [inline libs/infrastructure.py:50]: use shell.py ?

File: libs/infrastructure.py [hunk 2]
```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
...
+        _cmd_apply = ['terraform',
+                      'apply',
+                      '-var-file={}/{}'.format(self.tmp_root_path, TF_VARS_FILE),
+                      '-auto-approve',
+                      self.tf_working_dir]
+        yield "RUNNING: Actually creating cloud. This may take time... {}".format(_cmd_apply), None
+        process = Popen(_cmd_apply, cwd=self.cwd,
                         stdout=PIPE, stderr=PIPE)
         stdout, stderr = process.communicate()
         time.sleep(10)
```
Comment (ronenhamias) [inline libs/infrastructure.py:107]: time sleep is not a good idea as the process might take longer...

File: start_app.py [hunk 3]
```diff
@@ -175,26 +176,24 @@ def delete_cloud_provider_api(provider_type, name):
...
+    job = create_job(kube_cluster_create, app.logger, data).start()
+
+    return jsonify({'id': job.job_id})
+
 
     thread = Thread(target=kube_cluster_create,
-                    kwargs={'data': request.get_json()})
+                    kwargs={'data': request.get_json(), 'logger' : app.logger})
     thread.start()
     return {"kube_creation": "started"}
```
Comment (ronenhamias) [inline start_app.py]: should return a job id?
so status can be asked later....
{"id": "xxxx-xxxxxxx-xxxxxx-xxxxx-xxxx"}

File: libs/infrastructure.py [hunk 4]
```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
...
+        with open("{}/nodes_cm.yaml".format(self.tmp_root_path), "w") as nodes_cm:
             j2_env = Environment(loader=FileSystemLoader("/opt/templates/"),
                                  trim_blocks=True)
             gen_template = j2_env.get_template('nodes_cm.j2').render(aws_iam_role_eksnode_arn=role_arn)
             nodes_cm.write(gen_template)
 
-    def apply_node_auth_configmap(self):
+    def __apply_node_auth_configmap(self):
         self.generate_configmap()
         process = Popen(['kubectl', 'apply', "-f",
```
Comment (ronenhamias) [inline libs/infrastructure.py:74]: use shell.py?

File: libs/infrastructure.py [hunk 5]
```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
...
+        yield "RUNNING: Creating Terraform vars", None
+        self.__create_vars_file()
+        yield "RUNNING: Terraform vars created", None
+        _cmd_apply = ['terraform',
+                      'apply',
+                      '-var-file={}/{}'.format(self.tmp_root_path, TF_VARS_FILE),
+                      '-auto-approve',
+                      self.tf_working_dir]
+        yield "RUNNING: Actually creating cloud. This may take time... {}".format(_cmd_apply), None
+        process = Popen(_cmd_apply, cwd=self.cwd,
```
Comment (ronenhamias) [inline libs/infrastructure.py]: use shell.py?

File: libs/infrastructure.py [hunk 6]
```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
...
-    def set_aws_cli_config(self):
+    def __set_aws_cli_config(self):
+        result = 0
         process = Popen(["aws", "configure", "set", "aws_access_key_id",
                          "{}".format(self.aws_access_key)], stdout=PIPE, stderr=PIPE)
-        stdout, stderr = process.communicate()
-        process.wait()
-
+        result += process.wait()
         process = Popen(["aws", "configure",
```
Comment (ronenhamias) [inline libs/infrastructure.py:53]: use shell.py?

File: libs/infrastructure.py [hunk 7]
```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
...
                          "set", "aws_secret_access_key",
                          "{}".format(self.aws_secret_key)], stdout=PIPE, stderr=PIPE)
-        stdout, stderr = process.communicate()
-        process.wait()
+        result += process.wait()
 
         process = Popen(["aws", "configure",
```
Comment (ronenhamias) [inline libs/infrastructure.py:58]: use shell.py?

File: libs/infrastructure.py (hunk 2 above)
Comment (snripa) [reply libs/infrastructure.py:107]: sure, not this pr

File: libs/infrastructure.py (hunk 1 above)
Comment (snripa) [reply libs/infrastructure.py:50]: later, not this pr

File: libs/infrastructure.py (hunk 4 above)
Comment (snripa) [reply libs/infrastructure.py:74]: not this pr

File: libs/infrastructure.py (hunk 5 above)
Comment (snripa) [reply libs/infrastructure.py]: not this pr

File: libs/infrastructure.py (hunk 6 above)
Comment (snripa) [reply libs/infrastructure.py:53]: not this pr

**Pull request merged**